### PR TITLE
binary focal loss with logits implementation

### PR DESCRIPTION
**Pull request by xen0f0n**
Following issue #493 https://github.com/kornia/kornia/issues/493
I've implemented focal loss for the binary task. As the name suggests, input tensor should be the model logits.
As stated in the original [paper](https://arxiv.org/pdf/1708.02002.pdf), parameter `alpha` is assigned to the rare class. It's assumed that the rare class is labeled as "1".

### CODE REVIEW

**Comment by edgarriba** [comment]
to check local tests, from root of the project run: `make`
see: https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst#unit-testing

will go over the math the upcoming days

**Comment by edgarriba** [comment]
@xen0f0n thanks ! :)

**Comment by edgarriba** [comment]
@xen0f0n you have to rebase from master to sync heads
